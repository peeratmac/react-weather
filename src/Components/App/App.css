.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

nav {
  display: flex;
  justify-content: space-around;
  align-items: center;
  min-height: 10vh;
  background: #38ada9;
  color: white;
  font-size: 30px;
}

img {
  height: 50px;
}

.nav-links {
  width: 50%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  list-style: none;
}
