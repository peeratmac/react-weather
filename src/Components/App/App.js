import React, { Component } from 'react';
import {
  fetchPopularCities,
  fetchLattitudeLongtitude,
  fetchCity,
  fetchPopularWeather
} from '../../apiCalls';
import FormInput from '../../Components/FormInput/FormInput';
import Weather from '../../Components/Weather/Weather';
import WeatherCard from '../../Components/WeatherCard/WeatherCard';
import './App.css';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { loadCities, loadWeather } from '../../actions/index';

class App extends Component {
  componentDidMount = async () => {
    fetchPopularCities('NEW').then(cities => {
      console.log(cities);
      this.props.loadCities(cities);
      const IDs = cities.map(city => city.woeid);
      console.log(IDs);
      this.props.loadWeather(IDs);
    });
  };

  render() {
    console.log(this.props);
    return (
      <div className='app'>
        <h1>Weather App</h1>
        <FormInput />
        <WeatherCard />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  loadCities: cities => dispatch(loadCities(cities)),
  loadWeather: stationIDs => dispatch(loadWeather(stationIDs))
});

const mapStateToProps = state => ({
  cities: state.cities,
  stationIDs: state.stationIDs
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
