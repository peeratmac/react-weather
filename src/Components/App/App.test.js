import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('APP', () => {
  let wrapper;
  const mockLoadCities = jest.fn();
  const mockLoadWeather = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <App
        cities={''}
        stationIDs={''}
        loadCities={mockLoadCities}
        loadWeather={mockLoadWeather}
      />
    );
  });

  it('should match the App Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
