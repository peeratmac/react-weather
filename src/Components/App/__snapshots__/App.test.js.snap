// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the App Snapshot 1`] = `
<BrowserRouter>
  <div
    className="app"
  >
    <Nav />
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Favorites)",
            "type": [Function],
          }
        }
        path="/current-conditions"
      />
      <Route
        component={[Function]}
        path="/future-outlook"
      />
      <h1>
        Find Your Weather Station and Latitude Longitude
      </h1>
    </Switch>
  </div>
</BrowserRouter>
`;
